src/jemalloc.c

```diff
@@ -1931,22 +1930,24 @@ malloc_init_narenas(void) {
 	assert(ncpus > 0);
 
 	if (opt_percpu_arena != percpu_arena_disabled) {
+		bool cpu_count_is_deterministic =
+		    malloc_cpu_count_is_deterministic();
 		if (!cpu_count_is_deterministic) {
-			if (opt_narenas) {
-				malloc_write("<jemalloc>: Number of CPUs is not deterministic, "
-					"but narenas is set. Hope you not what you are doing and "
-					"you have set narenas to largest possible CPU ID.\n");
-				if (opt_abort) {
-					abort();
-				}
```

Comment:
I liked the warning when somebody uses `opt_narenas` + `opt_percpu_arena`. Maybe this could help someone detect an issue early.

Reply:
I considered that part too. But then I feel that `narenas` and `percpu_arena` are really conflicting options in principle -- if they are both set, it should be intentional.

The other thing is, I can't seem to find what to say in that case, especially since the setting might be a valid case. Then the warning msg will never go away, no matter how narenas is set. So I went with the approach to preserve the existing behavior (i.e. not causing any fuss for that case).

Reply:
Ok, that's a good way to think about it, too. I'm fine with it.